tests/test_rules.py

```diff
@@ -0,0 +1,280 @@
+    load the ``@handle_urls`` decorators from other modules/packages.
+    """
+    consume_modules("tests_extra")
+    rules = default_registry.get_rules()
+    page_objects = {po.use for po in rules}
+    assert any(["po_lib_sub_not_imported" in po.__module__ for po in page_objects])
+
+
+def test_registry_search_rules() -> None:
+    # param: use
+    rules = default_registry.search_rules(use=POTopLevel2)
+    assert len(rules) == 1
+    assert rules[0].use == POTopLevel2
+
+    # param: instead_of
+    rules = default_registry.search_rules(instead_of=POTopLevelOverriden2)
+    assert len(rules) == 1
+    assert rules[0].instead_of == POTopLevelOverriden2
+
+    # param: to_return
+    rules = default_registry.search_rules(to_return=Product)
+    assert len(rules) == 3
+    assert all([r for r in rules if r.use == Product])
+
+    # params: to_return and use
+    rules = default_registry.search_rules(to_return=Product, use=ImprovedProductPage)
+    assert len(rules) == 1
+    assert rules[0].to_return == Product
+    assert rules[0].use == ImprovedProductPage
+
+    # Such rules doesn't exist
+    rules = default_registry.search_rules(use=POModuleOverriden)
+    assert len(rules) == 0
+
+
+def test_registry_search_overrides_deprecation() -> None:
+    msg = "The 'search_overrides' method is deprecated. Use 'search_rules' instead."
+    with pytest.warns(DeprecationWarning, match=msg):
+        rules = default_registry.search_overrides(use=POTopLevel2)
+
+    # It should still work as usual
+    assert len(rules) == 1
+    assert rules[0].use == POTopLevel2
+
+
+def test_from_apply_rules() -> None:
+    rules = [
+        ApplyRule(
+            for_patterns=Patterns(include=["sample.com"]),
+            use=POTopLevel1,
+            instead_of=POTopLevelOverriden2,
+        )
+    ]
+
+    registry = PageObjectRegistry.from_apply_rules(rules)
+
+    assert registry.get_rules() == rules
+    assert default_registry.get_rules() != rules
+
+
+def test_from_override_rules_deprecation() -> None:
+    rules = [
+        ApplyRule(
+            for_patterns=Patterns(include=["sample.com"]),
+            use=POTopLevel1,
+            instead_of=POTopLevelOverriden2,
+        )
+    ]
+
+    msg = (
+        "The 'from_override_rules' method is deprecated. "
+        "Use 'from_apply_rules' instead."
+    )
+    with pytest.warns(DeprecationWarning, match=msg):
+        registry = PageObjectRegistry.from_override_rules(rules)
+
+    assert registry.get_rules() == rules
+    assert default_registry.get_rules() != rules
+
+
+def test_handle_urls_deprecation() -> None:
+    msg = (
+        "The 'overrides' parameter in @handle_urls is deprecated. Use the "
+        "'instead_of' parameter."
+    )
+    with pytest.warns(DeprecationWarning, match=msg):
+
+        @handle_urls("example.com", overrides=CustomProductPage)
+        class PageWithDeprecatedOverrides:
+            ...
```

Comment:
Yeah, it would, but it wouldn't affect how the other tests work due to the sequential reading of this test module. But that's a good point! as despite the deprecation warning, it should still be added successfully into the registry. https://github.com/scrapinghub/web-poet/pull/84/commits/1419f7a30aeeda580c9d3560de016bfde00b598f